src/foam/nanos/dig/DIG.js

```diff
@@ -138,6 +148,11 @@ foam.CLASS({
           query = true;
           url += "q=" + encodeURIComponent(q);
         }
+        if ( fields ) {
```

Comment:
fields will always be an array so this check will never be false. Maybe you want to check if fields.length > 0